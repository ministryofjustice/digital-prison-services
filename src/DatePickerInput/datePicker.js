import React, { Component } from 'react';
import Datetime from 'react-datetime';
import 'react-datetime/css/react-datetime.css';
import PropTypes from 'prop-types';

class DatePicker extends Component {
  constructor () {
    super();
    this.renderInput = this.renderInput.bind(this);
  }

  renderInput (props) {
    return (
      <div>
        <input id={this.props.inputId} className={this.props.className} name={this.props.name} {...props} readOnly/>
      </div>
    );
  }

  render () {
    const { shouldShowDay } = this.props;

    return (

      <Datetime
        className=""
        onChange={this.props.handleDateChange}
        timeFormat={false}
        isValidDate={shouldShowDay}
        locale="en-GB"
        dateFormat={"DD/MM/YYYY"}
        closeOnSelect
        strictParsing
        {...this.props}
        renderInput={this.renderInput}
      />);
  }
}


DatePicker.propTypes = {
  handleDateChange: PropTypes.func.isRequired,
  shouldShowDay: PropTypes.func.isRequired,
  className: PropTypes.string,
  name: PropTypes.string,
  title: PropTypes.string,
  inputId: PropTypes.string
};

export default DatePicker;
