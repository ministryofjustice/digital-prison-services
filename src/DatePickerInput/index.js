import React from 'react'
import moment from 'moment'
import PropTypes from 'prop-types'
import DatePicker from './datePicker'

const futureDateOnly = date =>
  date.isAfter(
    moment()
      .subtract(1, 'days')
      .startOf('day')
  )

const pastDateOnly = date =>
  date.isBefore(
    moment()
      .add(1, 'days')
      .startOf('day')
  )

const DatePickerInput = props => {
  const { additionalClassName, handleDateChange, label, value, inputId, error, futureOnly } = props

  const validDate = futureOnly ? futureDateOnly : pastDateOnly

  return (
    <DatePicker
      inputProps={{
        placeholder: 'Today',
        className: `datePickerInput form-control ${additionalClassName}`,
        label,
        error,
      }}
      name="date"
      shouldShowDay={validDate}
      title="Date"
      handleDateChange={handleDateChange}
      inputId={inputId}
      value={value}
    />
  )
}

DatePickerInput.propTypes = {
  handleDateChange: PropTypes.func.isRequired,
  value: PropTypes.string,
  inputId: PropTypes.string.isRequired,
  additionalClassName: PropTypes.string,
  label: PropTypes.string,
  error: PropTypes.string,
  futureOnly: PropTypes.bool,
}

DatePickerInput.defaultProps = {
  additionalClassName: '',
  label: '',
  error: '',
  value: '',
  futureOnly: false,
}

export default DatePickerInput
