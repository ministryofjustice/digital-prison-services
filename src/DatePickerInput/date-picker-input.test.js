import React from 'react'
import moment from 'moment'
import { shallow } from 'enzyme'
import DatePickerInput, { futureDateOnly, pastDateOnly } from './index'

describe('Date picker input', () => {
  describe('date picker input component', () => {
    it('should render correctly with default props', () => {
      const wrapper = shallow(
        <DatePickerInput
          handleDateChange={() => {}}
          inputId="date"
          label="date"
          placeholder="select date"
          value="2018-10-10"
        />
      )
      expect(wrapper.getElement().props.shouldShowDay).toBe(pastDateOnly)
      expect(wrapper).toMatchSnapshot()
    })

    it('should render with error', () => {
      const wrapper = shallow(<DatePickerInput error="there was an error" handleDateChange={() => {}} inputId="date" />)
      expect(wrapper).toMatchSnapshot()
    })

    it('should render with futureDates validator', () => {
      const wrapper = shallow(
        <DatePickerInput
          handleDateChange={() => {}}
          inputId="date"
          label="date"
          placeholder="select date"
          value="2018-10-10"
          futureOnly
        />
      )
      expect(wrapper.getElement().props.shouldShowDay).toBe(futureDateOnly)
    })
  })

  describe('past dates only', () => {
    it('should return true for yesterday', () => {
      const yesterday = moment().subtract(1, 'days')
      expect(pastDateOnly(yesterday)).toBe(true)
    })
    it('should return true for today', () => {
      expect(pastDateOnly(moment())).toBe(true)
    })
    it('should return true for tomorrow', () => {
      const tomorrow = moment().add(1, 'days')
      expect(pastDateOnly(tomorrow)).toBe(true)
    })
    it('should return false for the day after tomorrow', () => {
      const dayAfterTomorrow = moment().add(2, 'days')
      expect(pastDateOnly(dayAfterTomorrow)).toBe(false)
    })
  })

  describe('future dates only', () => {
    it('should return false for day before yesterday', () => {
      const yesterday = moment().subtract(2, 'days')
      expect(futureDateOnly(yesterday)).toBe(false)
    })
    it('should return true for today', () => {
      expect(futureDateOnly(moment())).toBe(true)
    })
    it('should return true for tomorrow', () => {
      const tomorrow = moment().add(1, 'days')
      expect(futureDateOnly(tomorrow)).toBe(true)
    })
  })
})
