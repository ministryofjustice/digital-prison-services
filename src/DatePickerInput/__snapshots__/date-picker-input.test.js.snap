// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date picker input date picker input component should render correctly with default props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatePickerInput
    additionalClassName=""
    error=""
    futureOnly={false}
    handleDateChange={[Function]}
    inputId="date"
    label="date"
    placeholder="select date"
    value="2018-10-10"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "handleDateChange": [Function],
      "inputId": "date",
      "inputProps": Object {
        "className": "datePickerInput form-control ",
        "error": "",
        "label": "date",
        "placeholder": "select date",
      },
      "name": "date",
      "shouldShowDay": [Function],
      "title": "Date",
      "value": "2018-10-10",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "handleDateChange": [Function],
        "inputId": "date",
        "inputProps": Object {
          "className": "datePickerInput form-control ",
          "error": "",
          "label": "date",
          "placeholder": "select date",
        },
        "name": "date",
        "shouldShowDay": [Function],
        "title": "Date",
        "value": "2018-10-10",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Date picker input date picker input component should render with error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatePickerInput
    additionalClassName=""
    error="there was an error"
    futureOnly={false}
    handleDateChange={[Function]}
    inputId="date"
    label=""
    placeholder="Today"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "handleDateChange": [Function],
      "inputId": "date",
      "inputProps": Object {
        "className": "datePickerInput form-control ",
        "error": "there was an error",
        "label": "",
        "placeholder": "Today",
      },
      "name": "date",
      "shouldShowDay": [Function],
      "title": "Date",
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "handleDateChange": [Function],
        "inputId": "date",
        "inputProps": Object {
          "className": "datePickerInput form-control ",
          "error": "there was an error",
          "label": "",
          "placeholder": "Today",
        },
        "name": "date",
        "shouldShowDay": [Function],
        "title": "Date",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
